source/Weapon.cpp

```diff
@@ -318,6 +328,20 @@ double Weapon::Range() const
 
 
 
+double Weapon::DamageDropoff(double distance) const
+{
+	double minDropoff = damageDropoff.first;
+	double maxDropoff = damageDropoff.second;
+	
+	if(distance <= minDropoff)
+		return 1.;
+	if(distance >= maxDropoff)
+		return damageDecay;
+	return (1 - damageDecay) / (minDropoff - maxDropoff) * (distance - minDropoff) + 1;
```

Comment:
This calculation looks correct, but is a bit difficult to read. Maybe splitting the calculation would improve readability? For example:
```
double dropoffFactor = (distance - minDropOff) / (maxDropoff - minDropOff);
return 1. - (1. - damageDecay) * dropoffFactor;
```

Reply:
I've put a comment here saying that the calculation is linear and split it up by defining the slope of the line separately.